src/libs/antares/study/parts/short-term-storage/container.cpp

```diff
@@ -25,20 +25,82 @@
 ** SPDX-License-Identifier: licenceRef-GPL3_WITH_RTE-Exceptions
 */
 
+#include <antares/logs.h>
+#include <filesystem>
+
 #include "container.h"
 
+#define SEP std::filesystem::path::preferred_separator
+
 namespace Antares::Data::ShortTermStorage
 {
+
 bool Container::validate() const
 {
     // TODO
     return false;
 }
+
+
 bool Container::createUnitsFromIniFile(const std::string& path)
 {
-    // TODO
+    for (const auto & file : std::filesystem::directory_iterator(path))
+    {
+        const std::string pathIni(file.path().generic_string() + SEP + "list.ini");
+        IniFile ini;
+        if (!ini.open(pathIni))
+            return false;
+
+        if (ini.firstSection)
+            return true;
+
+        for (auto* section = ini.firstSection; section; section = section->next)
+        {
+            Unit unit;
+            if (!unit.loadFromSection(*section))
+                return false;
+
+            storagesByIndex.push_back(unit);
+        }
+
+    }
+    return true;
+}
+
+bool Unit::loadFromSection(const IniFile::Section& section)
```

Comment:
we should put member functions of different classes in different source files, but it's probably already the plan